pageserver/src/lib.rs

```diff
@@ -129,6 +131,30 @@ impl PageServerConf {
             pg_distrib_dir: "".into(),
             auth_type: AuthType::Trust,
             auth_validation_public_key_path: None,
+            relish_storage_config: None,
         }
     }
 }
+
+#[derive(Debug, Clone)]
+pub enum RelishStorageConfig {
+    LocalFs(PathBuf),
```

Comment:
Thinking of `LocalFs` case, the plan to include it was for testing.
The more I consider it, the more I think it's a bit weird to have both this and a separate freezing logic for our layers.

Would it make more sense to extract the freezing logic to the `LocalFs` logic and use it always, while keeping the s3/whatnot options optional?

Reply:
I'm still thinking about the reasons we have this enum variant.
Initially, [in Discord](https://discord.com/channels/869525774699462656/882901114343788595/882904195215294537) we've planned to a storage interface that can be either s3 or local fs, the latter will be used for testing purposes, if I got that right.

But after the initial review discussion, looks like this fs wrapper now does exactly the same as the layer freezing does now, just differently implemented.
And if we would use that fs wrapper when testing, we'll be writing the same data twice.
It's useful if we want to compare those files, but otherwise might bring many issues or unnecessarily slow the system.

For now, I just want to understand what that new "local fs storage" functionality brings to us: maybe it was useful to prototype the trait interface and now we can throw it away and put a TODO on our current freezing code to extract it into that trait.

Reply:
> I'm still thinking about the reasons we have this enum variant.
> Initially, [in Discord](https://discord.com/channels/869525774699462656/882901114343788595/882904195215294537) we've planned to a storage interface that can be either s3 or local fs, the latter will be used for testing purposes, if I got that right.

Right, the point of "local fs storage" is testing. Being able to run all (or most) the same tests without a real S3 service running.

> But after the initial review discussion, looks like this fs wrapper now does exactly the same as the layer freezing does now, just differently implemented.
> And if we would use that fs wrapper when testing, we'll be writing the same data twice.
> It's useful if we want to compare those files, but otherwise might bring many issues or unnecessarily slow the system.

Perhaps it's not very useful yet, but I think it will come handy for testing restoring a page server from the data in S3, for example. And we might want to enhance it with more testing functionality later, like simulating failures in downloading/uploading files.

Another approach might be to use a simple S3 server implementation for testing. I used https://github.com/hlinnaka/ms3 earlier, that was the simplest I could find, but I had to modify it a bit to make it work. We might actually want to integrate something like that into our test suite anyway, to test our real S3 implementation. But still, I think a local fs implementation of this trait will come handy.